packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -121,6 +121,9 @@ function __experimentalBlockVariationTransforms( { blockClientId } ) {
 	// Check if each variation has a unique icon.
 	const hasUniqueIcons = useMemo( () => {
 		const variationIcons = new Set();
+		if ( ! variations ) {
```

Comment:
The expected behaviour when there are no variations is to fall back to `VariationsDropdown`, correct? In that case, we should make the logic more robust, because the patch currently only covers falsy values for `variations`, but not empty values like `variations: []`.

```suggestion
		if ( ! variations?.length ) {
```

Reply:
@mcsf, if there's no variations component rendering is skipped.

https://github.com/WordPress/gutenberg/blob/38aeca81a6338a17d503e0b97926a33aaf4ce63d/packages/block-editor/src/components/block-variation-transforms/index.js#L144-L145